src/test/java/analyzer/AnalyzerIntegrationTest.java

```diff
@@ -126,7 +126,26 @@ void needforspeed(String scenario) throws IOException {
         var path = Path.of("need-for-speed", scenario + ".java");
         var solution = new SolutionFromFiles("need-for-speed", SCENARIOS.resolve(path));
         var output = AnalyzerRoot.analyze(solution);
-      
+
+        Approvals.verify(serialize(output.analysis()), Approvals.NAMES.withParameters(scenario));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "ExemplarSolution",
+            "HardCodingLogLevels",
+            "NoReuseLogLevel",
+            "NoReuseMessage",
+            "NoReuseOfBothMethods",
+            "NotUsingSubstringOnLogLevel",
+            "NotUsingSubstringOnMessage",
```

Comment:
Let's add a test case where the solution doesn't use `substring` at all, so `NotUsingSubstringOnBothMethods`. Combined with my other comment I would expect the `use_substring_method` comment to only be added once in that case.